### ISSUE TITLE
Is it possible to turn on the high - contrast mode when testing WCAG SC 1.4.3 Contrast (Minimum)?

### ISSUE THREAD

**Issue by Zoey0li** [w3c/wcag#4290]
Is it possible to turn on the high - contrast mode when testing WCAG SC 1.4.3 Contrast (Minimum)? If not, then what is the function of the system's high-contrast mode? Didn't we design the high-contrast mode precisely to help people with visual impairments read the screen better?

**Comment by Zoey0li** [comment]
In addition, does the deep mode not belong to assistive technology? If I turn on the deep mode and pass the contrast test, can it also prove that the page has passed SC 1.4.3?

**Comment by patrickhlauke** [comment]
high contrast is AT that tries to override/force what an author has defined. the point of 1.4.3 is that the content itself should have sufficient contrast, because not every user agent/OS has the same type of "high contrast mode" adaptations, and a user should be able to comfortably see/read content without HAVING to first jump through hoops.

**Comment by patrickhlauke** [comment]
also, no idea what "deep mode" is...

**Comment by mraccess77** [comment]
High contrast mode is there to help people who need more than a minimum level of contrast. WCAG contrast requirements are meant as a minimum level for users so that specialized accessibility features are not required. Requiring folks to use high contrast mode when they just need a little is problematic. Use of high contrast is often too much for many folks causing issues. Some browsers like Chrome don't support platform contrast settings (to my knowledge) and the Chrome contrast extension is no longer supported.

**Comment by alastc** [comment]
Draft response:

<hr>

> Is it possible to turn on the high - contrast mode when testing WCAG SC 1.4.3 Contrast (Minimum)? 

High-contrast is not relevant to testing Contrast (min). The requirement is for the authored styles, rather than a user-override.

Assuming you meant "dark mode", it would depend how that is activated. If it is automatically activated then the all the colour combinations must meet Contrast (min). If it is a switch provided in the site's interface then either light or dark mode could be used to meet Contrast (min).

**Comment by Zoey0li** [comment]
We are currently testing the EU's accessibility directive. The standard EN 301549 requires that mobile phone apps also comply with the requirements of the Web Content Accessibility Guidelines (WCAG). When we were testing mobile phone apps, the manufacturer informed us that it is very difficult to design a mobile phone to meet Success Criterion (SC) 1.4.3 without turning on assistive technologies such as high contrast mode. In this situation, When testing SC 1.4.3, is it also necessary to turn off the high - contrast mode?

**Comment by mraccess77** [comment]
I'm not sure why it would be harder to meet sufficient contrast on a mobile device.  1.4.3 only requires a minimal level of contrast and not a special high contrast setting based on the user platform preferences.

**Comment by scottaohara** [comment]
sounds to me like this question isn't merely about high contrast themes and how they apply to websites - but rather a device's "increase contrast" settings which has a similar purpose, but often different results/adjustments based on the platform

**Comment by mbgower** [comment]
Relevant discussion https://github.com/w3c/wcag/issues/2909#issuecomment-1377975101

**Comment by mbgower** [comment]
Going to leave in draft for now, in case anyone wants to review/wordsmith, given our TF discussion on Mar 28

**Comment by alastc** [comment]
Updated **_draft_** response, potentially controversial:

<hr>

> Is it possible to turn on the high - contrast mode when testing WCAG SC 1.4.3 Contrast (Minimum)? 

High-contrast mode in Windows is not relevant to testing Contrast (min). The requirement is for the authored styles, rather than a user-override. (It's the same for other OSs where the mode overrides the authored styles.)

Assuming you meant "dark mode", it would depend how that is activated. If it is automatically activated then all the colour combinations must meet Contrast (min). If it is a switch provided in the site's interface then either light or dark mode could be used to meet Contrast (min).

For platforms which have an "increased contrast" setting, it needs to be widely available and known. For websites in general I don't think that would be the case, but if you can make a good case that it is available to your users then you could rely on it to meet 1.4.3.

For native apps, that is technically out of scope for this group. In terms of how it might be interpreted for EN 301 549 it may be similar to the previous point, you need to make a case that it's widely available to your users.

 > When we were testing mobile phone apps, the manufacturer informed us that it is very difficult to design a mobile phone to meet Success Criterion (SC) 1.4.3 without turning on assistive technologies such as high contrast mode.

We don't understand why it would be harder to meet contrast on a mobile app than a website. There is a good argument that contrast should be higher on a mobile device given the wider viewing conditions. Perhaps they are acknowledging that some of the default interfaces and components have low contrast?

**Comment by mraccess77** [comment]
@alastc when you say dark mode is activated automatically - do you mean if the platform supports dark mode and the site detects that platform setting and goes into a dark mode then the site would need to have sufficient contrast in that mode?  I just want to understand that scenario.  And I think we aren't saying dark mode has to be supported - but that if a site does offer dark mode and applies it automatically based on user platform settings then it would need to conform.

**Comment by fstrr** [comment]
> If it is automatically activated then all the colour combinations must meet Contrast (min). If it is a switch provided in the site's interface then either light or dark mode could be used to meet Contrast (min).

Does this also mean that if dark mode is automatically activated and there's also a switch mode to turn it off then that also means that either mode can be used to meet Contrast (Minimum)?

**Comment by TestPartners** [comment]
Yes, but the mechanism for switching colour schemes, and the path to the mechanism, must meet all WCAG success criteria. As far as I can tell, WCAG does not require that the switching mechanism is initially visible, in which case it could be hidden in a dropdown menu or a dialog. Although that would be a pretty stupid thing to do, I encountered it on a recent project.

**Comment by mbgower** [comment]
Based on discussion at today's Task Force call, we have rewritten the draft response to make it more succinct and directed, and opened up #4317 to tackle questions around page conformance

**Comment by detlevhfischer** [comment]
> When we were testing mobile phone apps, the manufacturer informed us that it is very difficult to design a mobile phone to meet Success Criterion (SC) 1.4.3 without turning on assistive technologies such as high contrast mode.

I am not sure the answer given to that is helpful. A very common problem in mobile iOS apps is the default contrast of toggle controls, but following the exception in 1.4.11 ("except for inactive components or where the appearance of the component is determined by the user agent and not modified by the author") and the equivalence of user agent for web content to OS settings for native apps, those might be exempt. It is also debatable that a high contrast setting is an assisitve technology.

**Comment by mbgower** [comment]
@detlevhfischer the issue opener brought up mobile in a [comment](https://github.com/w3c/wcag/issues/4290#issuecomment-2753302416), which I believe is why the response mentions it. 

The issue is specific to text contrast. When you mention toggle controls, are you talking about the label or the styling of the toggle itself? The latter seems to be for non-text contrast (and you cite 1.4.11), which I think is out of scope for this response?

Do you have suggestions on how you'd rephrase?

**Comment by mbgower** [comment]
@detlevhfischer I have edited the final paragraph to remove the following:
> There is an argument that contrast should be higher on a mobile device given the wider viewing conditions (e.g. viewing in direct sunlight). Perhaps they are acknowledging that some of the default interfaces and components have low contrast?

I've also added a link in the response to the MATF issue. Since I believe this has addressed your concern, I am marking the response as 'official' and marking this as closed. Feel free to reopen if you disagree.
